src/credential_provider.rs

```diff
@@ -31,6 +31,59 @@ pub struct CredentialProvider {
     pub cache_endpoint: String,
 }
 
+impl CredentialProvider {
+    /// Returns a builder to produce a Credential Provider using an API key stored in the specified
+    /// environment variable
+    ///
+    /// # Arguments
+    ///
+    /// * `env_var_name` - Name of the environment variable to read token from
+    /// # Examples
+    ///
+    /// ```
+    /// # tokio_test::block_on(async {
+    ///     use momento::CredentialProvider;
```

Comment:
Minor thing, but this spacing is preserved in the docs, so the example code will all be indented 4 extra spaces.

Reply:
ack